registry/tracecat_registry/base/core/http.py

```diff
@@ -2,16 +2,37 @@
 # XXX(WARNING): Do not import __future__ annotations from typing
 # This will cause class types to be resolved as strings
 
+import tempfile
 from typing import Annotated, Any, Literal, TypedDict
 
 import httpx
 from pydantic import Field, UrlConstraints
 
-from tracecat_registry import logger, registry
+from tracecat_registry import RegistrySecret, logger, registry, secrets
 
 RequestMethods = Literal["GET", "POST", "PUT", "DELETE"]
 JSONObjectOrArray = dict[str, Any] | list[Any]
 
+http_secret = RegistrySecret(
+    name="http",
```

Comment:
perhaps `ssl` is a more suitable name? given all the keys are `SSL_*`. It's fine if the secret name != the function/namespace name imo